src/playground-code-editor.ts

```diff
@@ -505,21 +506,31 @@ export class PlaygroundCodeEditor extends LitElement {
       }
     });
 
-    cm.on('renderLine', (_editorInstance: Editor, line, elt) => {
-      if (!this.lineWrapping) {
-        return;
-      }
+    this._indentingLine(cm);
 
-      const basePadding = 4;
-      const charWidth = cm.defaultCharWidth();
-      const tabSize = cm.getOption('tabSize') || basePadding;
-      const off = CodeMirror.countColumn(line.text, null, tabSize) * charWidth;
+    this._codemirror = cm;
+  }
 
-      elt.style.textIndent = `-${off}px`;
-      elt.style.paddingLeft = `${basePadding + off}px`;
-    });
+  private _indentingLine(cm: CodeMirror.Editor) {
+    if (this.lineWrapping) {
+      cm.on('renderLine', this._onRenderLine);
+    } else {
+      cm.off('renderLine', this._onRenderLine);
+    }
+  }
 
-    this._codemirror = cm;
+  private _onRenderLine(
+    _editorInstance: Editor,
+    line: CodeMirror.LineHandle,
+    elt: HTMLElement
+  ) {
+    const basePadding = 4;
+    const charWidth = _editorInstance.defaultCharWidth();
```

Comment:
What if the font size is different? I guess the indentation will always be fixed, regardless of the actual font size?

Reply:
![Screenshot 2022-04-04 at 23 52 10](https://user-images.githubusercontent.com/7101754/161645454-92660381-6dee-4736-9389-e337a592f943.png)
![Screenshot 2022-04-04 at 23 48 58](https://user-images.githubusercontent.com/7101754/161645460-842c0745-d6e6-45d2-a4f1-2f1285b84e6b.png)

Looks like it does okay with different fonts and font sizes. CodeMirror does have a note about defaultCharWidth.
> Note that for non-monospace fonts, this is mostly useless, and even for monospace fonts, non-ascii characters might have a different width

Doesn't seem too bad
![image](https://user-images.githubusercontent.com/7101754/161646605-4b587db5-a56a-4034-bea4-854373fe4b08.png)

Reply:
I wonder if we could use `em` or `ch` or a similar font-aware CSS unit? Then we might not need to ask codemirror for any font info at all.

https://developer.mozilla.org/en-US/docs/Web/CSS/length#syntax

Reply:
Using `ch` could most likely work for monospace fonts, which I am guessing most would be using a monospace font for the editor anyway. I can have a look into how this could be done.

I had a quick peek to see how CodeMirror was working it out https://github.com/codemirror/CodeMirror/blob/master/src/measurement/position_measurement.js#L607-L615 Looks like they try to get an average size and then cache it.

This is a nice article discussing the use of `ch` in a similar way https://meyerweb.com/eric/thoughts/2018/06/28/what-is-the-css-ch-unit/

Reply:
Cool, would you be able to try that out? Seems like it would be simpler, faster, and would dynamically adapt to the font.

Reply:
Yeah `ch` works better. I realised that there was an issue with the guttering as there is additional padding when line numbers are added. Due to the guttering needed to use `calc` for the padding left. Let me know what you think